CsharpExtras/Extensions/ArrayExtension2D.cs

```diff
@@ -170,6 +170,34 @@ public static int Count<TVal>(this TVal[,] array, Func<TVal, bool> checkerFuncti
             return resultArrayZeroBased;
         }
 
+        /// <summary>
+        /// Zips one or more arrays into a single array using an enumerable-based fold operation.
+        /// </summary>
+        /// <typeparam name="TVal">The type of values in this array</typeparam>
+        /// <typeparam name="TOther">The type of values in the enumerable of other arrays</typeparam>
+        /// <typeparam name="TResult">The type of values in the resultant array</typeparam>
+        /// <param name="zipper">A function which, given an element from this array and an enumerable of elements from 
+        /// the other arrays, returns a value in the resultant array.</param>
+        /// <param name="others">An enumerable of other arrays.</param>
+        /// <returns>A new array, the shape of which is the intersection of the shapes of this array and all the others, 
+        /// and the values of which are the result of applying the zipper across all values at the corresponding indices 
+        /// in all the arrays.</returns>
+        public static TResult[,] ZipEnum<TVal, TOther, TResult>(this TVal[,] array, Func<TVal, IEnumerable<TOther>, TResult> zipper, IEnumerable<TOther[,]> others)
+        {
+            int minRows = others.Select(o => o.GetLength(0)).UnionMin(array.GetLength(0).AsSingleton());

```

Comment:
Would it be possible to do the `.Union` first and then a `Min`?
Something like:

```
others.Select(o => o.GetLength(0)).Union(array.GetLength(0)).Min();
```

Or even:
```
others.Union(array).Select(o => o.GetLength(0)).Min();
```

It might be slightly clearer and more standard. The same applies to the other places that use this format.